### ISSUE TITLE
Cumulative results?

### ISSUE THREAD

**Issue by waiyeung1** [karolzak/boxdetect#20]
I have 2 documents which are almost identical except one is a shrunk a bit and a bit more grainy.

Boxdetect can get all the checkboxes on DocA but only some of the checkboxes on DocB, with this configuration:

    # important to adjust these values to match the size of boxes on your image
    cfg.width_range = [(30, 70)]
    cfg.height_range = [(30, 70)]

    # w/h ratio range for boxes/rectangles filtering
    cfg.wh_ratio_range = [(0.8, 1.2)]

    # num of iterations when running dilation tranformation (to engance the image)
    cfg.dilation_iterations = [1]
    cfg.dilation_kernel = [(1,1)]
 
When I attempt to capture more checkboxes in DocB with this configuration:

    # important to adjust these values to match the size of boxes on your image
    cfg.width_range = [(30, 70),(40, 70),(40, 70)]
    cfg.height_range = [(30, 70),(40, 70),(40, 70)]

    # w/h ratio range for boxes/rectangles filtering
    cfg.wh_ratio_range = [(0.8, 1.2),(0.8, 1.2),(0.8, 1.2)]

    # num of iterations when running dilation tranformation (to engance the image)
    cfg.dilation_iterations = [1,2,1]
    cfg.dilation_kernel = [(1,1),(2,2),(1,4)]

I'm finding that the checkboxes boxdetect originally captured in DocA are no longer captured, though the original configuration is in index 0 of the configuration.  Why is that?

**Comment by karolzak** [comment]
Hello @waiyeung1 
Is it possible for you to share these documents?

Overall if the doc B is almost identical but smaller and more grainy it may require a different configuration than doc A.  
E.g.: if for doc A, width range is `(30,70)` for doc B it should be smaller, like `(20,50)` (depending how much smaller is doc B. So for doc B I would try something like below:
```python
# important to adjust these values to match the size of boxes on your image
cfg.width_range = [(20, 60)]
cfg.height_range = [(20, 60)]

# w/h ratio range for boxes/rectangles filtering
cfg.wh_ratio_range = [(0.8, 1.2)]

# num of iterations when running dilation tranformation (to engance the image)
cfg.dilation_iterations = [1]
cfg.dilation_kernel = [(1,1)]
```

But without seeing the actual documents - it is just guessing

**Comment by waiyeung1** [comment]
Hi @karolzak,
Thanks for the quick response and suggestion.  Unfortunately, I can't share the documents.

I was able to determine the cause.  With the configuration, the missing checkboxes actually had 2 rectangles drawn around them that weren't merged when group_countours() was called.  As a result the checkbox wasn't considered a checkbox.  By passing in epsilon=0.2, the 2 contours were merged and the checkbox was determined to be a checkbox.

Cheers,
Wai
